### PR TITLE
[WIP] Refactor routes to use async

### LINKED ISSUE #907: Refactor routes to remove callbacks
This is an ongoing issue.

All routes should be refactored to have no callbacks, and instead use async/await

### PR DESCRIPTION
**Pull request by dekkerglen**
#Fixes #907 

This will likely cause many conflicts

### CODE REVIEW

**Comment by ruler501** [comment]
Why not move the bulkUpload function out? Reducing conflicts with my PR?

**Comment by RKelln** [comment]
Another option to consider with this refactor is to split the file up? cube, drafting, maybe, api, not sure the exact splits? Probably better as a separate PR after this tho.

File: routes/cube_routes.js [hunk 1]
```diff
@@ -259,81 +216,77 @@ router.post('/format/add/:id', ensureAuth, async (req, res) => {
...
+    }
+
+    let cube = await Cube.findOne(build_id_query(req.params.id));
+
+    cube.date_updated = Date.now();
+    cube.updated_string = cube.date_updated.toLocaleString('en-US');
+    cube = setCubeType(cube, carddb);
+
+    await cube.save();
+    const user = await User.findById(req.user._id);
```
[the remark this answers is not in the document]
Comment (dekkerglen) [reply routes/cube_routes.js]: No need, this route has ensureAuth

**Comment by dekkerglen** [comment]
find misses will be caught by the wrapping try/catch, so that's no issue.

**Pull request merged**